dap_job_quality/utils/spacy_keyword_search.py

```diff
@@ -0,0 +1,105 @@
+from dap_job_quality.getters.ojo_getters import get_ojo_sample
+from dap_job_quality.utils.text_cleaning import clean_text
+
+
+import pandas as pd
+import spacy
+from spacy.matcher import PhraseMatcher, Matcher
+import spacy
+from spacy import displacy
+from spacy.matcher import Matcher
+
+
+def get_matches(text, patterns, matcher, nlp):
+    """Takes text and returns spacy matches, based on the patterns.
+
+    Args:
+        text (str): A string of text
+        patterns (list): a list of spacy matcher patterns, following rule-based patterns: https://spacy.io/usage/rule-based-matching
+        matcher (matcher): a spacy matcher object
+
+    Returns:
+        _type_: spacy doc and list of matches (tuples of the match id, start span, end span)
+    """
+
+    for i in range(0, len(patterns)):
+        matcher.add(patterns[i]["subcategory"], patterns[i]["patterns"])
+
+    doc = nlp(text)
+    matches = matcher(doc)
+
+    return doc, matches
+
+
+def get_spans(doc, matches, nlp):
```

Comment:
Each span in the returned list contains: the label (eg "L&D"); the token indices of the start and end of the span; and the full sentence in which the span occurs - right? Could you add a little bit to the docstring to it clear what the format of each element in `collapsed_list` consists of?